src/components/Button/Button.stories.tsx

```diff
@@ -0,0 +1,123 @@
+import { StoryFn, Meta } from "@storybook/react";
+import { Button, Props } from "./Button";
+import { MdAccountCircle } from "react-icons/md";
+import "../../styles/globals.css";
+
+export default {
+  title: "Components/Button",
+  component: Button,
+  description: "The reusable button component",
+  argTypes: {
+    size: {
+      control: {
+        type: "inline-radio",
+        options: ["regular", "small"],
+      },
+    },
+    onClick: { action: "clicked" },
+    disabled: { control: "boolean" },
+    variant: {
+      control: {
+        type: "inline-radio",
+        options: ["primary", "primarySmall", "secondary", "secondarySmall"],
+      },
+    },
+    iconPosition: {
+      control: {
+        type: "inline-radio",
+        options: ["left", "right"],
+      },
+    },
+  },
+} as Meta;
+
+const Template: StoryFn<Props> = (args) => <Button {...args} />;
+
+export const Primary = Template.bind({});
+Primary.args = {
+  variant: "primary",
+  children: "Primary Button",
+  disabled: false,
+  href: "/",
+};
+
+export const Secondary = Template.bind({});
+Secondary.args = {
+  variant: "secondary",
+  children: "Secondary Button",
+  disabled: false,
+  href: "/",
+};
+
+export const PrimaryWithIcon = Template.bind({});
```

Comment:
Acho que você pode ter apenas uma story, porque é parametrizável.
O icon você pode fazer uma combobox com alguns icons pré prontos para ilustrar como funciona o button com o Icon e a position você pode programar isso no storybook pra inverter a ordem (pensando na utilização do button usando composition).
O storybook é altamente customizável dá pra criar sua story to jeito que quiser, e os controls também tem uma enorme customização. Acho que você vai curtir fazer isso, não é muito demorado. Procure por controls na documentação do storybook e veja como você pode fazer algo ultra parametrizável.